fitsio/tests/test_bit_datatype.rs

```diff
@@ -38,5 +38,5 @@ fn test_writing_bit_data_type() {
 
     let table_hdu = f.hdu("DATA").unwrap();
     let flags: Vec<u32> = dbg!(table_hdu.read_col(&mut f, "BITMASK").unwrap());
-    assert_eq!(flags.len(), 64);
+    assert!(data.iter().zip(&flags).all(|(a, b)| a == b));
```

Comment:
minor: we should probably check that `flags` has the correct length, since `zip` will stop after the length of the shortest array. If for example, `flags` is empty (length 0) then this assertion will always pass.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b7080864f72f7b07bcb77154e6c4ccaa
```suggestion
    assert_eq!(flags.len(), 64);
    assert!(data.iter().zip(&flags).all(|(a, b)| a == b));
```

**But**: thanks for improving this test assertion! Only checking the length was not really testing the implementation correctly!

Reply:
Oh! I totally missed that side-effect of my switching the test. Thanks for catching it! But yeah, changing the test actually did find bugs. As did using values that used the full width of the type.